test/testdata/infer/generics/t_class_of.rb

```diff
@@ -0,0 +1,212 @@
+# typed: true
+
+# TODO(jez) Another consideration for all of this: having the constant literal
+# `A` have type `T.class_of(A)[T.untyped]` is kind of like allowing `A` to
+# simply not implement an abstract type, which is just like how it's bad that
+# singleton class methods can be abstract.
+#
+# Maybe we should require non-fixed, invariant type members to be marked
+# `abstract!`?
+#
+
+class Module
+  include T::Sig
+end
+
+class A
+  extend T::Sig
+  extend T::Generic
+  abstract!
+
+  X = type_template
+
+  sig {abstract.returns(X)}
+  def self.foo; end
+
+  sig {returns(Integer)}
+  def instance_method; 0; end
+end
+
+class ChildA < A
+  extend T::Sig
+  extend T::Generic
+
+  X = type_template
+
+  sig {override.returns(X)}
+  def self.foo; raise; end
+end
+
+class ChildAFixedString < A
+  extend T::Sig
+  extend T::Generic
+
+  X = type_template {{fixed: String}}
+
+  sig {override.returns(X)}
+  def self.foo; ''; end
+end
+
+
+sig do
+  type_parameters(:U)
+    .params(klass: T.class_of(A)[A, Integer])
+    .void
+end
+def example0(klass)
+  x = klass.foo
+  T.reveal_type(x) # error: `Integer`
+  a = klass.new
+  T.reveal_type(a) # error: `A`
+end
+
+
+sig do
+  type_parameters(:U)
+    .params(klass: T.class_of(A)[Integer])
+    #                            ^^^^^^^ error: Wrong number of type parameters for `T.class_of(A)`
+    #                            ^^^^^^^ error: Wrong number of type parameters for `T.class_of(A)`
+    #                            ^^^^^^^ error: `Integer` is not a subtype of upper bound of type member `::<Class:A>::<AttachedClass>`
+    #                            ^^^^^^^ error: `Integer` is not a subtype of upper bound of type member `::<Class:A>::<AttachedClass>`
```

Comment:
(for future improvement, not on this branch) Do you think it would be worth hiding the `is not a subtype of ... <AttachedClass>` if there is a type parameter mismatch? It's likely that means the user forgot to supply the attached class (or some other type parameter), so this error is just noise until the mismatch is fixed.

Reply:
Not sure. We can chat about it. It likely ties into whatever our eventual decision around whether this could ever be defaulted is.

Reply:
By comparison, we report two errors for the corresponding error with normal params/args type mismatches

https://sorbet.run/#%23%20typed%3A%20true%0Aextend%20T%3A%3ASig%0A%0Asig%20%7Bparams%28x%3A%20Integer%2C%20y%3A%20String%29.void%7D%0Adef%20foo%28x%2C%20y%29%0Aend%0A%0Afoo%28''%29

Reply:
Fair enough, we'd definitely want the behaviour of type param mismatches and normal params mismatches to be the same (whether that's silence when the count is wrong or show always is something to chat about later).